src/Modules/Fx.asm

```diff
@@ -303,6 +303,25 @@ play_beep_melody_loop_next_item:
     inc  DPTR
     djnz Temp5, play_beep_melody_loop
 
+play_beep_melody_wait:
+    ; Read the melody wait setting and wait a number of ms before exiting
+    ; This is used to make playback on multiple ESCs end at the same time.
+    mov  DPTR, #Eep_Pgm_Melody_End_Wait
+    clr  A
+    movc A, @A+DPTR
+    mov  Temp3, A                       ; wait_ms (hi byte)
+
+    inc  DPTR
+    clr  A
+    movc A, @A+DPTR
+    mov  Temp2, A                       ; wait_ms (lo byte)
+
+    ; Exit if long wait time (60+ seconds), to safeguard against uninitialized flash
+    cpl  A
+    jz   play_beep_melody_exit          ; Exit if Temp3 is 255
```

Comment:
A is loaded with Temp2.
Should it be loaded with Temp3 before cpl instead?
Could it be better to compare be sure it is below a threshold?

Reply:
Realistically we only want to protect against waiting 0xFF in they hi byte - so yes, cpl is in the wrong place.

I am not sure if an absolute comparison would make sense at this point. I mean it is very unlikely that a hi byte of 0xFF makes sense... Sure we can check for if hi byte is bigger than 0x0F - that should be more than enough to sync it. Lot's of melodies are synced now "manually" anyway.